packages/compiler/src/server/serverlib.ts

```diff
@@ -339,6 +342,45 @@ export function createServer(host: ServerHost): Server {
     }
   }
 
+  async function updateImportsOnFileMovedOrRenamed(param: {
+    oldFilePath: string;
+    newFilePath: string;
+    openedFilePath: string;
+  }): Promise<string[]> {
+    const result: string[] = [];
+
+    const resultCompile = await compileService.compile({
+      uri: fileService.getURL(normalizePath(param.openedFilePath)),
+    });
+    if (resultCompile === undefined) {
+      return [];
+    }
+
+    const { program } = resultCompile;
+    const oldFileName = getBaseFileName(param.oldFilePath);
+    for (const diagnostic of program.diagnostics) {
+      if (diagnostic.code === "import-not-found") {
+        const target = diagnostic.target as Node;
+        if (
+          target.parent &&
+          target.kind === SyntaxKind.ImportStatement &&
+          target.path.value.includes(oldFileName)
+        ) {
+          const filePath = target.parent.file.path;
+          const text = target.parent.file.text;
+          result.push(getBaseFileName(filePath));
+          // The value of 'target.path.value' is the value of imports, so the value of imports needs to be replaced here.
+          const replaceText = getRelativePathByComparePaths(filePath, param.newFilePath);
+          if (replaceText.length > 0) {
+            await compilerHost.writeFile(filePath, text.replace(target.path.value, replaceText));
```

Comment:
the replace feels dangerous, it's better to replace by using the target position to avoid replacing user's code incorrectly.